diffusion/algorithms/pin_diffusion_rng.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 MosaicML Diffusion authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Algorithm to pin diffusion process noise."""
+
+from typing import Any, Dict
+
+import torch
+from composer.core import Algorithm, Event, State
+from composer.loggers import Logger
+from composer.utils import dist
+
+from diffusion.models import StableDiffusion
+
+
+class PinDiffusionRNG(Algorithm):
+    """Algorithm to pin diffusion process noise."""
+
+    def __init__(self) -> None:
+        self.seed = 0
+        self.train_rng_state = None
```

Comment:
Should be typed, implicit optionals are bad.

Reply:
Fixed!